partition_processor.go

```diff
@@ -297,50 +294,58 @@ func (pp *PartitionProcessor) Start(ctx context.Context) error {
 	return nil
 }
 
+func (pp *PartitionProcessor) stopping() <-chan struct{} {
+	return pp.state.WaitForStateMin(PPStateStopping)
+}
+
 // Stop stops the partition processor
 func (pp *PartitionProcessor) Stop() error {
 	pp.log.Debugf("Stopping")
 	defer pp.log.Debugf("... Stopping done")
 	pp.state.SetState(PPStateStopping)
-	defer pp.state.SetState(PPStateIdle)
-	errs := new(multierr.Errors)
+	defer pp.state.SetState(PPStateStopped)
```

Comment:
Idle will never be set again. Is this intended?

Reply:
it's set at the beginning as an initial state. But once it leaves the idle state it will never go back to avoid reusing a partition-processor. It's a run-once-thing, that's just safer to implement that way.